### PR TITLE
HIVE-25482: Add option to enable connectionLeak detection for Hikari …

### PR DESCRIPTION
**Pull request by avpash43**
…datasource

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java [hunk 1]
```diff
@@ -67,6 +67,7 @@ public DataSource create(Configuration hdpConfig) throws SQLException {
     config.setJdbcUrl(driverUrl);
     config.setUsername(user);
     config.setPassword(passwd);
+    config.setLeakDetectionThreshold(3600*1000);
```
Comment (rbalamohan) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java:70]: Can you get this value via Configuration object? (as in, make it configurable value)

Comment (avpash43) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java]: Fixed it

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 2]
```diff
@@ -602,6 +602,13 @@ public static ConfVars getMetaConf(String name) {
     CONNECTION_USER_NAME("javax.jdo.option.ConnectionUserName",
         "javax.jdo.option.ConnectionUserName", "APP",
         "Username to use against metastore database"),
+    CONNECTION_LEAK_DETECTION_THRESHOLD("javax.jdo.option.ConnectionLeakDetectionThreshold",
```
Comment (rbalamohan) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:605]: 1. This impacts HiveMetaStore. Can you refer to MetaStoreConf?
2. Add a test case in TestDataSourceProviderFactory?

Comment (avpash43) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:605]: "Can you refer to MetaStoreConf?" - I didn't catch you. What do you mean? Just create constant or what?

**Comment by rbalamohan** [comment]
Thanks @avpash43 . Changes LGTM. Some tests are failing, you can view the link and retrigger the tests. Let me retrigger the test now.

+1 pending tests.

**Comment by rbalamohan** [comment]
I have retriggered the tests (link: http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2610/4/pipeline)

**Comment by avpash43** [comment]
Thank you @rbalamohan , I will try to trigger tests also.

**Comment by avpash43** [comment]
@rbalamohan , all tests has been finished successfully. Can we make merge?

**Pull request merged**